packages/client-sdk-nodejs/src/auth-client-props.ts

```diff
@@ -0,0 +1,18 @@
+import {CredentialProvider} from '.';
+import {Configuration} from './config/configuration';
+
+export interface AuthClientProps {
+  /**
+   * Configuration settings for the auth client
+   */
+  configuration: Configuration;
+  /**
+   * controls how the client will get authentication information for connecting to the Momento service
+   */
+  credentialProvider: CredentialProvider;
```

Comment:
I'm 90% sure that we don't want this field in our final public API for this.  That credential provider is for providing API tokens, which AFAIK will not be used for many/most of the APIs that people consume the AuthClient for.

Once this PR is merged we will either need to be happy committing to what is in it as a public API for AuthClient, since it would be included in the next node.js release, or we will be blocked on doing any more node.js releases until we file follow-up PRs to solidify the API.  My pref would be to try to get it all sorted in this PR - apologies that this is landing on you Ben, we were planning to do this stuff but you started working on this before we had time.

Reply:
cc: @rlinehan

Reply:
this is probably the biggest UX/API issue we need to sort out.  We need the control plane endpoint for this client, but not all of the APIs in this client require an API token.  We don't currently have a mechanism for a user to get access to the control plane endpoint without parsing it from an API token.

From a practical perspective this client should only require the endpoint uri to its constructor, but from a UX perspective I am on the fence about how they would discover that and get it into their code.

Reply:
We have the same issue for momento cli atm, we plan to just hardcode endpoints for now w/ custom option. Seems like we don't have a real solution without a endpoint discovery system, which it seems we are going to need really soon here. 

For now, seems best that we supply a helper `(aws/gcp, region): endpoint` in node sdk and have the `AuthClient` only accept a url.